internal/cli/action/customize.go

```diff
@@ -50,76 +51,95 @@ func Customize(ctx *cli.Context) error {
 
 	logger.Info("Customizing image started")
 
-	tmpDir, err := vfs.TempDir(fs, "", "customize-")
+	imagePath, configPath := resolveOutputPaths(args)
+
+	output, err := config.NewOutput(fs, "", configPath)
 	if err != nil {
-		logger.Error("Creating temporary working directory failed")
+		logger.Error("Creating working directory failed")
 		return err
 	}
 
 	defer func() {
-		logger.Debug("Cleaning up temporary working directory", tmpDir)
-		rmErr := fs.RemoveAll(tmpDir)
-		if rmErr != nil {
-			logger.Error("Cleaning up temporary working directory '%s' failed: %v", tmpDir, rmErr)
+		if rmErr := output.Cleanup(fs); rmErr != nil {
+			logger.Error("Cleaning up working directory failed: %v", rmErr)
 		}
 	}()
 
-	def, err := digestCustomizeDefinition(fs, args)
+	def, err := digestCustomizeDefinition(fs, args, imagePath)
 	if err != nil {
 		logger.Error("Digesting image definition from customize flags failed")
 		return err
 	}
 
-	outDir := config.OutputDir(tmpDir)
 	ctxCancel, cancelFunc := signal.NotifyContext(ctx.Context, syscall.SIGTERM, syscall.SIGINT)
 	defer cancelFunc()
 
-	customizeRunner, err := setupCustomizeRunner(ctxCancel, system, args, outDir)
+	customizeRunner, err := setupCustomizeRunner(ctxCancel, system, args, output)
 	if err != nil {
 		logger.Error("Setting up customization runner failed")
 		return err
 	}
 
-	if err = customizeRunner.Run(ctxCancel, def, outDir, args.Local); err != nil {
+	if err = customizeRunner.Run(ctxCancel, def, output, args.Local); err != nil {
 		logger.Error("Customizing installer media failed")
 		return err
 	}
 
 	return nil
 }
 
+func resolveOutputPaths(args *cmd.CustomizeFlags) (imagePath, configPath string) {
+	imagePath = args.OutputPath
```

Comment:
Would a `nil` check be worth it here?

Reply:
I don't seem to understand -- the args variable cannot be nil no matter what we do.